### ISSUE TITLE
kubernetes cri-containerd image error on reboot

### ISSUE THREAD

**Issue by w9n** [linuxkit/linuxkit#2632]
**Description**
cri-containerd kubernetes version doesnt succefully reboot.

**Steps to reproduce the issue:**
```
KUBE_MASTER_AUTOINIT="" ./boot.sh
<wait and check for kubernetes>
poweroff
KUBE_PRESERVE_STATE=true ./boot.sh
```

**Describe the results you received:**
no task in the k8s.io namespace start, lots of missing image errors in the logs.

ctr -n k8s.io images list
```
ERRO[0000] failed calculating size for image docker.io/weaveworks/weave-kube:2.0.4  error="content digest sha256:1ad426d140e5e15b6584f3dbad0e98084993f02ad95e3cba2086b0b054a04033: not found"
ERRO[0000] failed calculating size for image docker.io/weaveworks/weave-kube@sha256:cddb2620620481271763d39c797b2f0b970d18bc5e98ffb7151a3c5f8a923324  error="content digest sha256:1ad426d140e5e15b6584f3dbad0e98084993f02ad95e3cba2086b0b054a04033: not found"
ERRO[0000] failed calculating size for image docker.io/weaveworks/weave-npc:2.0.4  error="content digest sha256:0dad21d6c11e2aad989c0359bd7d2f60470aede01a186dc651c645542dffc28b: not found"
ERRO[0000] failed calculating size for image docker.io/weaveworks/weave-npc@sha256:676b3a815e2cb2477e6d4ed3c6699c3d3a9c23e30b8f5e76af8e5c0b5030d525  error="content digest sha256:0dad21d6c11e2aad989c0359bd7d2f60470aede01a186dc651c645542dffc28b: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/etcd-amd64:3.0.17  error="content digest sha256:243830dae7dd6ff78859fa1d66098a89e2951a9e95af4ef4d4d2c03d97975771: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/etcd-amd64@sha256:d83d3545e06fb035db8512e33bd44afb55dea007a3abd7b17742d3ac6d235940  error="content digest sha256:243830dae7dd6ff78859fa1d66098a89e2951a9e95af4ef4d4d2c03d97975771: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5  error="content digest sha256:459944ce8cc4f08ebade5c05bb884e4da053d73e61ec6afe82a0b1687317254c: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64@sha256:46b933bb70270c8a02fa6b6f87d440f6f1fce1a5a2a719e164f83f7b109f7544  error="content digest sha256:459944ce8cc4f08ebade5c05bb884e4da053d73e61ec6afe82a0b1687317254c: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.5  error="content digest sha256:512cd7425a731bee1f2a3e4c825fc1cfe516c8b7905874f24bee6da12801d663: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-kube-dns-amd64@sha256:1a3fc069de481ae690188f6f1ba4664b5cc7760af37120f70c86505c79eea61d  error="content digest sha256:512cd7425a731bee1f2a3e4c825fc1cfe516c8b7905874f24bee6da12801d663: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5  error="content digest sha256:fed89e8b4248a788655d528d96fe644aff012879c782784cd486ff6894ef89f6: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/k8s-dns-sidecar-amd64@sha256:9aab42bf6a2a068b797fe7d91a5d8d915b10dbbc3d6f2b10492848debfba6044  error="content digest sha256:fed89e8b4248a788655d528d96fe644aff012879c782784cd486ff6894ef89f6: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-apiserver-amd64:v1.8.1  error="content digest sha256:e24ccd46aab5467efa0e6f855309b198d02107b1c7e532d1c9318284129547eb: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-apiserver-amd64@sha256:e1f4066f2f8e0cf975d8427842d95c86fd872c239e9ed2534f83121ebdce10a1  error="content digest sha256:e24ccd46aab5467efa0e6f855309b198d02107b1c7e532d1c9318284129547eb: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-controller-manager-amd64:v1.8.1  error="content digest sha256:08514038038e098e40981673e7e374ba2d62a24d321afc6e8122a07c50e10dd1: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-controller-manager-amd64@sha256:d9a1ab41ba39939511077ce2da44774daafaaecc6460a62358c6e0428831b7d2  error="content digest sha256:08514038038e098e40981673e7e374ba2d62a24d321afc6e8122a07c50e10dd1: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-proxy-amd64:v1.8.1  error="content digest sha256:30643046b829689fdc2cb1308d35a14dc085bfc46e6a8a411869ca0616c4650d: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-proxy-amd64@sha256:58fd496116bf3d2790279f9c44040bdfa71791aba7a7a1b5ca35025b5a032b9f  error="content digest sha256:30643046b829689fdc2cb1308d35a14dc085bfc46e6a8a411869ca0616c4650d: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-scheduler-amd64:v1.8.1  error="content digest sha256:e9999818cef22dfd2942e22fa878281701abd844d9cd8f4c674171c7234dee4f: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/kube-scheduler-amd64@sha256:fbb29b579b8f20075073987023caa146783d85eac66808c0860f99187fc02454  error="content digest sha256:e9999818cef22dfd2942e22fa878281701abd844d9cd8f4c674171c7234dee4f: not found"
ERRO[0000] failed resolving platform for image gcr.io/google_containers/pause:3.0  error="content digest sha256:320aedbfaad3d719b5e6a711834d2d06e2b0a172ce6fdadf84c640b6136914ed: not found"
ERRO[0000] failed resolving platform for image sha256:08514038038e098e40981673e7e374ba2d62a24d321afc6e8122a07c50e10dd1  error="content digest sha256:08514038038e098e40981673e7e374ba2d62a24d321afc6e8122a07c50e10dd1: not found"
ERRO[0000] failed resolving platform for image sha256:243830dae7dd6ff78859fa1d66098a89e2951a9e95af4ef4d4d2c03d97975771  error="content digest sha256:243830dae7dd6ff78859fa1d66098a89e2951a9e95af4ef4d4d2c03d97975771: not found"
ERRO[0000] failed resolving platform for image sha256:30643046b829689fdc2cb1308d35a14dc085bfc46e6a8a411869ca0616c4650d  error="content digest sha256:30643046b829689fdc2cb1308d35a14dc085bfc46e6a8a411869ca0616c4650d: not found"
ERRO[0000] failed resolving platform for image sha256:320aedbfaad3d719b5e6a711834d2d06e2b0a172ce6fdadf84c640b6136914ed  error="content digest sha256:320aedbfaad3d719b5e6a711834d2d06e2b0a172ce6fdadf84c640b6136914ed: not found"
ERRO[0000] failed calculating size for image sha256:3e2e9472eefc1a5fd0b727d8cc87afbb6352e888e6f818ed453bb893dfd9a6a8  error="content digest sha256:1ad426d140e5e15b6584f3dbad0e98084993f02ad95e3cba2086b0b054a04033: not found"
ERRO[0000] failed resolving platform for image sha256:459944ce8cc4f08ebade5c05bb884e4da053d73e61ec6afe82a0b1687317254c  error="content digest sha256:459944ce8cc4f08ebade5c05bb884e4da053d73e61ec6afe82a0b1687317254c: not found"
ERRO[0000] failed resolving platform for image sha256:512cd7425a731bee1f2a3e4c825fc1cfe516c8b7905874f24bee6da12801d663  error="content digest sha256:512cd7425a731bee1f2a3e4c825fc1cfe516c8b7905874f24bee6da12801d663: not found"
ERRO[0000] failed calculating size for image sha256:a33806b1a7dec8aea3fe1a27286bbcad37c342d7ee23c3d587102911c23fe0f6  error="content digest sha256:0dad21d6c11e2aad989c0359bd7d2f60470aede01a186dc651c645542dffc28b: not found"
ERRO[0000] failed resolving platform for image sha256:e24ccd46aab5467efa0e6f855309b198d02107b1c7e532d1c9318284129547eb  error="content digest sha256:e24ccd46aab5467efa0e6f855309b198d02107b1c7e532d1c9318284129547eb: not found"
ERRO[0000] failed resolving platform for image sha256:e9999818cef22dfd2942e22fa878281701abd844d9cd8f4c674171c7234dee4f  error="content digest sha256:e9999818cef22dfd2942e22fa878281701abd844d9cd8f4c674171c7234dee4f: not found"
ERRO[0000] failed resolving platform for image sha256:fed89e8b4248a788655d528d96fe644aff012879c782784cd486ff6894ef89f6  error="content digest sha256:fed89e8b4248a788655d528d96fe644aff012879c782784cd486ff6894ef89f6: not found"
REF                                                                                                                            TYPE                                                      DIGEST                                                                  SIZE      PLATFORM                          LABELS 
docker.io/weaveworks/weave-kube:2.0.4                                                                                          application/vnd.docker.distribution.manifest.list.v2+json sha256:cddb2620620481271763d39c797b2f0b970d18bc5e98ffb7151a3c5f8a923324 2.8 KiB   linux/amd64,linux/arm,linux/arm64 -      
docker.io/weaveworks/weave-kube@sha256:cddb2620620481271763d39c797b2f0b970d18bc5e98ffb7151a3c5f8a923324                        application/vnd.docker.distribution.manifest.list.v2+json sha256:cddb2620620481271763d39c797b2f0b970d18bc5e98ffb7151a3c5f8a923324 2.8 KiB   linux/amd64,linux/arm,linux/arm64 -      
docker.io/weaveworks/weave-npc:2.0.4                                                                                           application/vnd.docker.distribution.manifest.list.v2+json sha256:676b3a815e2cb2477e6d4ed3c6699c3d3a9c23e30b8f5e76af8e5c0b5030d525 2.2 KiB   linux/amd64,linux/arm,linux/arm64 -      
docker.io/weaveworks/weave-npc@sha256:676b3a815e2cb2477e6d4ed3c6699c3d3a9c23e30b8f5e76af8e5c0b5030d525                         application/vnd.docker.distribution.manifest.list.v2+json sha256:676b3a815e2cb2477e6d4ed3c6699c3d3a9c23e30b8f5e76af8e5c0b5030d525 2.2 KiB   linux/amd64,linux/arm,linux/arm64 -      
gcr.io/google_containers/etcd-amd64:3.0.17                                                                                     application/vnd.docker.distribution.manifest.v2+json      sha256:d83d3545e06fb035db8512e33bd44afb55dea007a3abd7b17742d3ac6d235940 43.0 MiB  -                                 -      
gcr.io/google_containers/etcd-amd64@sha256:d83d3545e06fb035db8512e33bd44afb55dea007a3abd7b17742d3ac6d235940                    application/vnd.docker.distribution.manifest.v2+json      sha256:d83d3545e06fb035db8512e33bd44afb55dea007a3abd7b17742d3ac6d235940 43.0 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.5                                                                    application/vnd.docker.distribution.manifest.v2+json      sha256:46b933bb70270c8a02fa6b6f87d440f6f1fce1a5a2a719e164f83f7b109f7544 10.8 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64@sha256:46b933bb70270c8a02fa6b6f87d440f6f1fce1a5a2a719e164f83f7b109f7544   application/vnd.docker.distribution.manifest.v2+json      sha256:46b933bb70270c8a02fa6b6f87d440f6f1fce1a5a2a719e164f83f7b109f7544 10.8 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-kube-dns-amd64:1.14.5                                                                         application/vnd.docker.distribution.manifest.v2+json      sha256:1a3fc069de481ae690188f6f1ba4664b5cc7760af37120f70c86505c79eea61d 12.5 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-kube-dns-amd64@sha256:1a3fc069de481ae690188f6f1ba4664b5cc7760af37120f70c86505c79eea61d        application/vnd.docker.distribution.manifest.v2+json      sha256:1a3fc069de481ae690188f6f1ba4664b5cc7760af37120f70c86505c79eea61d 12.5 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-sidecar-amd64:1.14.5                                                                          application/vnd.docker.distribution.manifest.v2+json      sha256:9aab42bf6a2a068b797fe7d91a5d8d915b10dbbc3d6f2b10492848debfba6044 10.8 MiB  -                                 -      
gcr.io/google_containers/k8s-dns-sidecar-amd64@sha256:9aab42bf6a2a068b797fe7d91a5d8d915b10dbbc3d6f2b10492848debfba6044         application/vnd.docker.distribution.manifest.v2+json      sha256:9aab42bf6a2a068b797fe7d91a5d8d915b10dbbc3d6f2b10492848debfba6044 10.8 MiB  -                                 -      
gcr.io/google_containers/kube-apiserver-amd64:v1.8.1                                                                           application/vnd.docker.distribution.manifest.v2+json      sha256:e1f4066f2f8e0cf975d8427842d95c86fd872c239e9ed2534f83121ebdce10a1 28.4 MiB  -                                 -      
gcr.io/google_containers/kube-apiserver-amd64@sha256:e1f4066f2f8e0cf975d8427842d95c86fd872c239e9ed2534f83121ebdce10a1          application/vnd.docker.distribution.manifest.v2+json      sha256:e1f4066f2f8e0cf975d8427842d95c86fd872c239e9ed2534f83121ebdce10a1 28.4 MiB  -                                 -      
gcr.io/google_containers/kube-controller-manager-amd64:v1.8.1                                                                  application/vnd.docker.distribution.manifest.v2+json      sha256:d9a1ab41ba39939511077ce2da44774daafaaecc6460a62358c6e0428831b7d2 24.7 MiB  -                                 -      
gcr.io/google_containers/kube-controller-manager-amd64@sha256:d9a1ab41ba39939511077ce2da44774daafaaecc6460a62358c6e0428831b7d2 application/vnd.docker.distribution.manifest.v2+json      sha256:d9a1ab41ba39939511077ce2da44774daafaaecc6460a62358c6e0428831b7d2 24.7 MiB  -                                 -      
gcr.io/google_containers/kube-proxy-amd64:v1.8.1                                                                               application/vnd.docker.distribution.manifest.v2+json      sha256:58fd496116bf3d2790279f9c44040bdfa71791aba7a7a1b5ca35025b5a032b9f 27.5 MiB  -                                 -      
gcr.io/google_containers/kube-proxy-amd64@sha256:58fd496116bf3d2790279f9c44040bdfa71791aba7a7a1b5ca35025b5a032b9f              application/vnd.docker.distribution.manifest.v2+json      sha256:58fd496116bf3d2790279f9c44040bdfa71791aba7a7a1b5ca35025b5a032b9f 27.5 MiB  -                                 -      
gcr.io/google_containers/kube-scheduler-amd64:v1.8.1                                                                           application/vnd.docker.distribution.manifest.v2+json      sha256:fbb29b579b8f20075073987023caa146783d85eac66808c0860f99187fc02454 12.4 MiB  -                                 -      
gcr.io/google_containers/kube-scheduler-amd64@sha256:fbb29b579b8f20075073987023caa146783d85eac66808c0860f99187fc02454          application/vnd.docker.distribution.manifest.v2+json      sha256:fbb29b579b8f20075073987023caa146783d85eac66808c0860f99187fc02454 12.4 MiB  -                                 -      
gcr.io/google_containers/pause:3.0                                                                                             application/vnd.oci.image.manifest.v1+json                sha256:27865b15496841a89083c94d4c51ede5e77be8b837ab8eb737dd968453f147c5 304.9 KiB -                                 -      
sha256:08514038038e098e40981673e7e374ba2d62a24d321afc6e8122a07c50e10dd1                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:d9a1ab41ba39939511077ce2da44774daafaaecc6460a62358c6e0428831b7d2 24.7 MiB  -                                 -      
sha256:243830dae7dd6ff78859fa1d66098a89e2951a9e95af4ef4d4d2c03d97975771                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:d83d3545e06fb035db8512e33bd44afb55dea007a3abd7b17742d3ac6d235940 43.0 MiB  -                                 -      
sha256:30643046b829689fdc2cb1308d35a14dc085bfc46e6a8a411869ca0616c4650d                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:58fd496116bf3d2790279f9c44040bdfa71791aba7a7a1b5ca35025b5a032b9f 27.5 MiB  -                                 -      
sha256:320aedbfaad3d719b5e6a711834d2d06e2b0a172ce6fdadf84c640b6136914ed                                                        application/vnd.oci.image.manifest.v1+json                sha256:27865b15496841a89083c94d4c51ede5e77be8b837ab8eb737dd968453f147c5 304.9 KiB -                                 -      
sha256:3e2e9472eefc1a5fd0b727d8cc87afbb6352e888e6f818ed453bb893dfd9a6a8                                                        application/vnd.docker.distribution.manifest.list.v2+json sha256:cddb2620620481271763d39c797b2f0b970d18bc5e98ffb7151a3c5f8a923324 2.8 KiB   linux/amd64,linux/arm,linux/arm64 -      
sha256:459944ce8cc4f08ebade5c05bb884e4da053d73e61ec6afe82a0b1687317254c                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:46b933bb70270c8a02fa6b6f87d440f6f1fce1a5a2a719e164f83f7b109f7544 10.8 MiB  -                                 -      
sha256:512cd7425a731bee1f2a3e4c825fc1cfe516c8b7905874f24bee6da12801d663                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:1a3fc069de481ae690188f6f1ba4664b5cc7760af37120f70c86505c79eea61d 12.5 MiB  -                                 -      
sha256:a33806b1a7dec8aea3fe1a27286bbcad37c342d7ee23c3d587102911c23fe0f6                                                        application/vnd.docker.distribution.manifest.list.v2+json sha256:676b3a815e2cb2477e6d4ed3c6699c3d3a9c23e30b8f5e76af8e5c0b5030d525 2.2 KiB   linux/amd64,linux/arm,linux/arm64 -      
sha256:e24ccd46aab5467efa0e6f855309b198d02107b1c7e532d1c9318284129547eb                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:e1f4066f2f8e0cf975d8427842d95c86fd872c239e9ed2534f83121ebdce10a1 28.4 MiB  -                                 -      
sha256:e9999818cef22dfd2942e22fa878281701abd844d9cd8f4c674171c7234dee4f                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:fbb29b579b8f20075073987023caa146783d85eac66808c0860f99187fc02454 12.4 MiB  -                                 -      
sha256:fed89e8b4248a788655d528d96fe644aff012879c782784cd486ff6894ef89f6                                                        application/vnd.docker.distribution.manifest.v2+json      sha256:9aab42bf6a2a068b797fe7d91a5d8d915b10dbbc3d6f2b10492848debfba6044 10.8 MiB  -                                 -      
```
**Describe the results you expected:**
A running master

**Additional information you deem important (e.g. issue happens only occasionally):**
lots of content files in the ```/var/lib/containerd/io.containerd.content.v1.content/``` disappear after second boot. 
tested with bigger disk and more ram.

**Comment by ijc** [comment]
Anything interesting in `/var/log/*`? especially the `kubelet` and `cri-containerd` logs.

The fact that `/var/lib` is apprently being nuked is suspicious, maybe things aren't really being preserved or the disk is somehow getting reformatted?

**Comment by ijc** [comment]
I did an isolated (non-kube, just base LinuxKit) test of `linuxkit/format:be811ded880af37a916f23bc9ddb274026ccc514` and `linuxkit/mount:ef2e2106212b2d333b548821191f04da984ef48f` and it looks like the disk is being preserved correctly.

I've also just done a test with `linuxkitprojects/cri-containerd:afb7c111a40788b1e085856fbddacf8a878385e7` and `linuxkitprojects/kubernetes:718c9c7a47c518622d41ec0e8afeeacda3a363d6` which should correspond to v1.8.0, and I see something like reported above, which suggests this is not related to the bump to v1.8.1. It could be from the v1.7.x->v1.8.0 transition I suppose, or perhaps a newer containerd or cri-containerd is needed. e.g. I'm wondering if something like https://github.com/kubernetes-incubator/cri-containerd/issues/303 might be related (this exact case shouldn't be a problem in the repro I'm seeing since I am waiting till the system is fully up -- i.e. there are no inprogress pulls to be aborted by the reboot).

**Comment by ijc** [comment]
tested `cri-containerd` master (698f0ea2ae09) and `containerd` master (edf4114d5507) with the same result, so this doesn't appear to be fixed with those newer components.

**Comment by ijc** [comment]
The containerd garbage collector (introduced in `v1.0.0-beta.1`) seems to be nuking some required blobs. I'm just gathering enough information to turn into a containerd issue.

**Comment by ijc** [comment]
Filed as https://github.com/containerd/containerd/issues/1674

**Comment by ijc** [comment]
I think this is fixed by #2699
